haystack_experimental/dataclasses/chat_message.py

```diff
@@ -158,24 +158,34 @@ def is_from(self, role: ChatRole) -> bool:
         return self._role == role
 
     @classmethod
-    def from_user(cls, text: str) -> "ChatMessage":
+    def from_user(
+        cls,
+        text: str,
+        meta: Optional[Dict[str, Any]] = None,
+    ) -> "ChatMessage":
         """
         Create a message from the user.
 
         :param text: The text content of the message.
+        :param meta: Additional metadata associated with the message.
```

Comment:
In the current implementation of `ChatMessage` in Haystack, metadata are only supported in messages with assistant role.
So I replicated this in the experimental dataclass.
Do you think it is more consistent to allow metadata in `ChatMessage` for each possible role?